spanner/cloud-client/backup_sample_test.py

```diff
@@ -0,0 +1,125 @@
+
+
+def unique_instance_id():
+    """ Creates a unique id for the database. """
+    return 'test-instance-{}'.format(''.join(random.choice(
+        string.ascii_lowercase + string.digits) for _ in range(5)))
+
+
+def unique_database_id():
+    """ Creates a unique id for the database. """
+    return 'test-db-{}'.format(''.join(random.choice(
+        string.ascii_lowercase + string.digits) for _ in range(5)))
+
+
+def unique_backup_id():
+    """ Creates a unique id for the backup. """
+    return 'test-backup-{}'.format(''.join(random.choice(
+        string.ascii_lowercase + string.digits) for _ in range(5)))
+
+
+INSTANCE_ID = unique_instance_id()
+DATABASE_ID = unique_database_id()
+RESTORE_DB_ID = unique_database_id()
+BACKUP_ID = unique_backup_id()
+
+
+@pytest.fixture(scope='module')
+def spanner_instance():
+    spanner_client = spanner.Client()
+    instance_config = '{}/instanceConfigs/{}'.format(
+        spanner_client.project_name, 'regional-us-central1')
+    instance = spanner_client.instance(INSTANCE_ID, instance_config)
+    op = instance.create()
+    op.result(30)  # block until completion
+    yield instance
+    instance.delete()
+
+
+@pytest.fixture(scope='module')
+def database(spanner_instance):
+    """ Creates a temporary database that is removed after testing. """
+    db = spanner_instance.database(DATABASE_ID)
+    db.create()
+    yield db
+    db.drop()
+
+
+def test_create_backup(capsys, database):
+    backup_sample.create_backup(INSTANCE_ID, DATABASE_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Backup " in out
+    assert (BACKUP_ID + " of size 0 bytes was created at ") in out
+
+
+def test_restore_database(capsys):
+    backup_sample.restore_database(INSTANCE_ID, RESTORE_DB_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Database " in out
+    assert (DATABASE_ID + " restored to ") in out
+    assert (RESTORE_DB_ID + " from backup ") in out
+    assert (BACKUP_ID + ".") in out
+
+
+def test_update_backup(capsys):
+    backup_sample.update_backup(INSTANCE_ID, BACKUP_ID)
+    out, _ = capsys.readouterr()
+    assert "Backup " in out
+    assert (BACKUP_ID + " expire time was updated from ") in out
+    assert " to " in out
+
+
+def test_list_backups(capsys, spanner_instance):
+    backup_sample.list_backups(INSTANCE_ID)
+    out, _ = capsys.readouterr()
+    backup_name = spanner_instance.name + "/backups/" + BACKUP_ID
+    assert "All backups:\n" + backup_name in out
+    assert (
+        "All backups with backup name containing \"users\":\n"
+        "All backups with database name containing \"bank\":\n"
+        "All backups with expire_time before \"2019-10-18T02:56:53Z\":\n"
+        "All backups with backup size more than 1000 bytes:\n"
+        "All backups created after \"2019-10-18T02:56:53Z\" and are READY:\n"
+    ) in out
+    assert (
+        "All backups created after \"2019-10-18T02:56:53Z\" and are READY:\n" +
+        backup_name
+    ) in out
```

Comment:
I recommend adding a check to see if there's at least one backup returned. Otherwise it could be that we printed out the message but it didn't retrieve any backups.

Reply:
`backup_name` is constructed above and cannot be an empty string. If no backups are returned, the assertion would fail.